src/Heartbeat/Sensor.php

```diff
@@ -124,4 +125,20 @@ protected function _getNonCachedStatus(): Status
      * @return mixed The sensor status.
      */
     abstract protected function _getStatus();
+
+    /**
+     * Get the given setting or an optional default value
+     *
+     * @param string $name The name of the setting to retrieve.
+     * @param null|mixed $default The optional default value, if the setting is not set.
+     * @return string The name of the connection to check.
+     * @todo Cover by a test.
+     */
+    protected function _getSetting(string $name, $default = null): string
```

Comment:
@ravage84 I'm not sure how to test this.
Maybe it should be public? Or maybe you have an idea on how to test it.

Reply:
No, it doesn't need to. Sure it would make testing easier, but code doesn't need to change just for the sake of testability.

This can be tested by a test dummy. A class that extends ``Sensor`` and then implements a public method which simply calls ``_getSettings()`` and returns it.
This public method can be called in the test case and its return value be assserted. The to-be-asserted value can be put into the configuration, beforehand, naturally.

Reply:
On a second thought, we might want to move this method to the ``Config`` class and make it public. This way the code is where it should be AND is easier testable... Let's discuss this tomorow.